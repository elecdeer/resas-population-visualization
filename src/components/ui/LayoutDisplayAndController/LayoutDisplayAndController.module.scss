@use "src/styles/variables";

.layout-display-and-controller {
	display: grid;
	grid-template-columns: 1fr 1fr;
	margin: 8px;
	grid-gap: 8px;

  .display {
		order: 1;
  }

	.controller {
		order: 2;
	}

	// mobile
	@include variables.media-query(sm) {
		grid-template-columns: 1fr;

		.display {
			order: 2;
		}

		.controller {
			order: 1;
		}
	}
}

