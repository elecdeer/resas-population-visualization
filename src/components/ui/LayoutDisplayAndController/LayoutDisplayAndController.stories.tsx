import {
  LayoutDisplayAndController,
  LayoutDisplayAndControllerProps,
} from "./LayoutDisplayAndController";

export default {
  title: "ui/LayoutDisplayAndController",
  component: LayoutDisplayAndController,
};

export const Default = (args?: LayoutDisplayAndControllerProps) => (
  <LayoutDisplayAndController
    {...args}
    controller={<div style={{ background: "#ff9966" }}>Controller</div>}
    display={<div style={{ background: "#33ccff" }}>Display</div>}
  />
);
