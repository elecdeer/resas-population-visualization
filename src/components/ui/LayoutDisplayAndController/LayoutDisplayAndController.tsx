import React from "react";
import styles from "./LayoutDisplayAndController.module.scss";

export type LayoutDisplayAndControllerProps = {
  display: React.ReactElement;
  controller: React.ReactElement;
};

export const LayoutDisplayAndController: React.VFC<LayoutDisplayAndControllerProps> =
  ({ controller, display }) => {
    return (
      <div className={styles["layout-display-and-controller"]}>
        <div className={styles["controller"]}>{controller}</div>
        <div className={styles["display"]}>{display}</div>
      </div>
    );
  };
