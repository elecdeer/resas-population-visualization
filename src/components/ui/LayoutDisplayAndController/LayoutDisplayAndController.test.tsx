import { LayoutDisplayAndController } from "./LayoutDisplayAndController";
import { render } from "@testing-library/react";

test("LayoutDisplayAndController Component Snapshot Test", () => {
  const { asFragment } = render(<LayoutDisplayAndController />);
  expect(asFragment()).toMatchSnapshot();
});