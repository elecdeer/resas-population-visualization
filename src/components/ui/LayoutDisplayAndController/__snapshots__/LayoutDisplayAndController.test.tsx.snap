// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutDisplayAndController Component Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="layout-display-and-controller"
  >
    <div
      class="controller"
    />
    <div
      class="display"
    />
  </div>
</DocumentFragment>
`;
